moveit_core/robot_state/test/robot_state_benchmark.cpp

```diff
@@ -0,0 +1,144 @@
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the Willow Garage nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Author: Robert Haschke */
+#include <moveit_resources/config.h>
+#include <moveit/robot_model/robot_model.h>
+#include <moveit/robot_state/robot_state.h>
+#include <urdf_parser/urdf_parser.h>
+#include <boost/filesystem.hpp>
+#include <gtest/gtest.h>
+#include <fstream>
+#include <sstream>
+#include <algorithm>
+#include <ctime>
+#include <ctype.h>
+
+class PR2 : public testing::Test
+{
+protected:
+  std::string readFileToString(boost::filesystem::path path) const
+  {
+    std::string file_string;
+    std::fstream file(path.string().c_str(), std::fstream::in);
+    if (file.is_open())
+    {
+      std::string line;
+      while (file.good())
+      {
+        std::getline(file, line);
+        file_string += (line + "\n");
+      }
+      file.close();
+    }
+    return file_string;
+  }
+
+  void SetUp() override
+  {
+    boost::filesystem::path res_path(MOVEIT_TEST_RESOURCES_DIR);
+    const std::string urdf = this->readFileToString(res_path / "pr2_description/urdf/robot.xml");
+    const std::string srdf = this->readFileToString(res_path / "pr2_description/srdf/robot.xml");
+
+    urdf::ModelInterfaceSharedPtr parsed_urdf(urdf::parseURDF(urdf));
+    if (!parsed_urdf)
+      throw std::runtime_error("Cannot parse URDF.");
+
+    srdf::ModelSharedPtr parsed_srdf(new srdf::Model());
+    bool srdf_ok = parsed_srdf->initString(*parsed_urdf, srdf);
+    if (!srdf_ok)
+      throw std::runtime_error("Cannot parse URDF.");
+
+    model_.reset(new robot_model::RobotModel(parsed_urdf, parsed_srdf));
```

Comment:
In the future you should use @BryceStevenWilley 's new testing framework as documented here:
https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html